src/Build.UnitTests/Scanner_Tests.cs

```diff
@@ -115,6 +116,21 @@ public void WhitespaceProperty()
             Assert.Equal("IllFormedPropertyWhitespaceInCondition", lexer.GetErrorResource());
         }
 
+        [Fact]
+        public void WhitespacePropertyOptOutWave17()
+        {
+            using TestEnvironment env = TestEnvironment.Create();
+            env.SetChangeWave(ChangeWaves.Wave17_0);
+
+            Scanner lexer = new Scanner("$(x )", ParserOptions.AllowProperties);
+            AdvanceToScannerError(lexer);
+            Assert.Null(lexer.UnexpectedlyFound);
+
+            lexer = new Scanner("$( x)", ParserOptions.AllowProperties);
+            AdvanceToScannerError(lexer);
+            Assert.Null(lexer.UnexpectedlyFound);
```

Comment:
```suggestion
            using (TestEnvironment env = TestEnvironment.Create())
            {
                env.SetChangeWave(ChangeWaves.Wave17_0);

                Scanner lexer = new Scanner("$(x )", ParserOptions.AllowProperties);
                AdvanceToScannerError(lexer);
                Assert.Null(lexer.UnexpectedlyFound);

                lexer = new Scanner("$( x)", ParserOptions.AllowProperties);
                AdvanceToScannerError(lexer);
                Assert.Null(lexer.UnexpectedlyFound);
            }
```

Having `TestEnvironment` call `Dispose` resets state for CI builds.

Reply:
The previous way actually calls Dispose, too, when env goes out of scope, and I think it's technically recommended, although it isn't something we use in MSBuild particularly.

Reply:
I can change it back to the old syntax if you folks prefer.

Reply:
So long as dispose is called I'm okay with its current state 👍